utils/clipboard/clipboard.go

```diff
@@ -0,0 +1,62 @@
+package clipboard
+
+import (
+	"crypto/sha256"
+	"crypto/subtle"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strconv"
+
+	otherClipboard "github.com/atotto/clipboard"
+)
+
+const (
+	hashLen = 4 // in bytes
+)
+
+func CopyWithClearing(text string, clearAfter Duration) error {
+	if clearAfter.Seconds() > 0 {
+		// Clear the clipboard after 'clearAfter'
+		_ = launchClearer(text, clearAfter)
+	}
+	return otherClipboard.WriteAll(text)
+}
+
+func ClearIfMatch(match string) error {
+	// We're about to clear the clipboard
+	// Ensure that we still have the expected contents in it
+	// This avoids clearing other data which have been copied in the meantime
+	current, err := otherClipboard.ReadAll()
+	if err != nil {
+		return fmt.Errorf("failed to read current clipboard content: %s", err)
```

Comment:
imo it doesn't make sense to pre-define helper errors such as this. Also we can't really do it in this particular case as it wraps another error.

Reply:
I see your point. The reason I prefer pre-defining errors like this is because it allows us to explicitly handle those error cases later.

This is fine for now and we can potentially revisit it when we implement real logging.